### PR TITLE
Combine time-dep rotation maps

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

We have two different rotation maps in two separate files. One for 2D and one for 3D. This PR combines these maps into one `Rotation<Dim>` map that functions the same as the previous two for their respective dimensions.

Also the internals of QuaternionFunctionOfTime were changed to accommodate these changes. Essentially, previously the internal `PiecewisePolynomial` stored omega as a function of time. Now it stores the angle of rotation as a function of time (with omega being its first derivative). The main reason for this is because the control error for rotation (which will be added later) corresponds to an angle, not an angular velocity. Thus we want to update the nth derivative of a FunctionOfTime that represents an angle, not omega.

### Upgrade instructions

If you want to use the 3D rotation map that uses quaternions, use `Rotation<3>` now instead of `Rotation3D`.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.hpp [hunk 1]
```diff
@@ -146,6 +146,11 @@ class QuaternionFunctionOfTime : public FunctionOfTime {
       const QuaternionFunctionOfTime<LocalMaxDeriv>& lhs,
       const QuaternionFunctionOfTime<LocalMaxDeriv>& rhs);
 
+  template <size_t LocalMaxDeriv>
+  friend std::ostream& operator<<(  // NOLINT(readability-redundant-declaration
```
Comment (wthrowe) [inline src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.hpp]: Close paren.

File: tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_Rotation.cpp [hunk 2]
```diff
@@ -11,120 +11,227 @@
...
   using FoftPtr = std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>;
 
   const std::string f_of_t_name{"rotation_angle"};
-  const std::array<DataVector, deriv_order + 1> init_func{
-      {{1.0}, {-2.0}, {2.0}, {0.0}}};
   std::unordered_map<std::string, FoftPtr> f_of_t_list{};
-  f_of_t_list[f_of_t_name] =
-      std::make_unique<Polynomial>(t, init_func, final_time + dt);
+  std::array<DataVector, deriv_order + 1> init_func{};
+  Approx custom_approx = Approx::custom().epsilon(1.0).scale(1.0);
```
Comment (wthrowe) [inline tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_Rotation.cpp:152]: Put a comment here because it looks bad but isn't.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by knelli2** [comment]
Squashed directly.

**Review by wthrowe — APPROVED** [review]

**Comment by nilsvu** [comment]
@knelli2 Please make sure to put the upgrade instructions between the `<!-- UPGRADE_INSTRUCTIONS -->` fences in the PR description, so they get parsed correctly :) (I fixed it in this PR and in #3612)

**Pull request merged**